lib/vendors/axis-communications/sv_vendor_axis_communications.c

```diff
@@ -872,7 +890,17 @@ sv_vendor_axis_communications_set_attestation_report(signed_video_t *sv,
       sv->onvif = onvif_media_signing_create(sv_codec_to_ms_codec(sv->codec));
     }
   }
-
+  if (sv->sign_data) {
+    if (sv->onvif) {
+      // Set the signing key pair if sign_data is not available
+      char *private_key = get_private_key_from_sv(sv);
+      size_t private_key_size = strlen(private_key);
+      MediaSigningReturnCode onvif_status = onvif_media_signing_set_signing_key_pair(sv->onvif,
```

Comment:
I suggest that you avoid `MediaSigningReturnCode` and use only `SignedVideoReturnCode`/`svrc_t`

Also, what if ONVIF returns an error. Is it fine to return that error or are there other options?